src/Functions/parseDateTime.cpp

```diff
@@ -1035,6 +1035,37 @@ namespace
                 return cur;
             }
 
+            static Pos mysqlMicrosecond(Pos cur, Pos end, const String & fragment, DateTime & /*date*/)
+            {
+                checkSpace(cur, end, 6, "mysqlMicrosecond requires size >= 6", fragment);
+
+                Pos start = cur;
+                auto check_is_number = [&](Pos pos)
+                {
+                    if (*pos < '0' || *pos > '9')
+                        throw Exception(
+                            ErrorCodes::CANNOT_PARSE_DATETIME,
+                            "Unable to parse fragment '{}' from '{}' because '{}'' is not a number ",
+                            fragment,
+                            std::string_view(start, end),
+                            *cur);
+                };
+
+                check_is_number(cur);
+                ++cur;
+                check_is_number(cur);
+                ++cur;
+                check_is_number(cur);
+                ++cur;
+                check_is_number(cur);
+                ++cur;
+                check_is_number(cur);
+                ++cur;
+                check_is_number(cur);
+                ++cur;
+                return cur;
```

Comment:
where do you set date?

Reply:
The output type (`DateTime`) supports no fractional seconds, therefore we just parse the input (and check at least that we read numbers) but ignore it (for details, see the PR message)